File: fullstack-onlineshop/onlineshopapi/src/main/java/msg/onlineshopapi/dto/ProductDTO.java
package msg.onlineshopapi.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import msg.onlineshopapi.domain.enums.Category;

import java.math.BigDecimal;
import java.util.UUID;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class ProductDTO {
    
    private UUID id;
    private String name;
    private Category category;
    private String imageUrl;
    private BigDecimal price;
    private String description;
    private SupplierDTO supplier;
}

File: fullstack-onlineshop/onlineshopui/src/app/core/types/dtos/product.dto.ts
export type ProductCategoryDto = {
    id: string;
    name: string;
    description: string;
};

export type SupplierDto = {
    id: string;
    name: string;
    contactEmail: string;
    phone: string;
};

export type ProductDto = {
    id: string;
    name: string;
    description: string;
    price: number;
    weight: number;
    category: ProductCategoryDto;
    supplier: SupplierDto;
    imageUrl: string;
};

export type CreateProductRequest = Omit<ProductDto, 'id' | 'category' | 'supplier'> & {
    categoryId: string;
    supplierId: string;
};

export type UpdateProductRequest = Partial<ProductDto> & { categoryId?: string; supplierId?: string };
